resources/osl_systemd_unit_drop_in.rb

```diff
@@ -0,0 +1,60 @@
+# Based on https://github.com/facebook/chef-cookbooks/blob/master/cookbooks/fb_systemd/resources/override.rb
+resource_name 'osl_systemd_unit_drop_in'
+provides 'osl_systemd_unit_drop_in'
+unified_mode true
+
+default_action :create
+
+property :override_name, String, name_property: true
+property :unit_name, String, required: true
+property :content, [String, Hash], required: false
+property :triggers_reload, [true, false], default: true
+property :instance, kind_of: String, default: 'system'
+
+action_class do
+  def get_override_dir
+    "/etc/systemd/#{new_resource.instance}/#{new_resource.unit_name}.d"
+  end
+end
+
+action :create do
+  directory get_override_dir do
+    recursive true
+  end
+
+  file ::File.join(get_override_dir, "#{new_resource.override_name}.conf") do
+    content to_ini(new_resource.content).prepend("# Generated by Chef\n")
+    if new_resource.triggers_reload
+      notifies :run, 'execute[systemctl daemon-reload]', :immediately
+    end
+    verify { to_ini(new_resource.content) }
+  end
+
+  execute 'systemctl daemon-reload' do
+    action :nothing
+  end
+end
+
+action :delete do
+  if ::Dir.exist?(get_override_dir)
+    file ::File.join(get_override_dir, "#{new_resource.override_name}.conf") do
+      action :delete
+      if new_resource.triggers_reload
+        notifies :run, 'execute[systemctl daemon-reload]', :immediately
+      end
```

Comment:
Using a guard here is cleaner:
```suggestion
      notifies :run, 'execute[systemctl daemon-reload]', :immediately if new_resource.triggers_reload
```

Reply:
👍